biggerquery/monitoring.py

```diff
@@ -0,0 +1,164 @@
+
+
+def format_rfc3339(dt):
+    return dt.isoformat("T") + "Z"
+
+
+def get_start_time():
+    start_time = datetime.datetime.utcnow() - datetime.timedelta(minutes=5)
+    return format_rfc3339(start_time)
+
+
+def get_now_rfc3339():
+    return format_rfc3339(datetime.datetime.utcnow())
+
+
+def metric_exists(client, project_resource, metric_type):
+    response = api_list_metrics(client, project_resource, metric_type)
+    try:
+        return response['metricDescriptors'] is not None
+    except KeyError:
+        logger.warning('Metric not found: {}'.format(metric_type))
+        return False
+
+
+def wait_for_metric(client, project_resource, metric_type):
+    retries_left = 10
+    while not metric_exists(client, project_resource, metric_type) and retries_left:
+        logger.info('Waiting for metric: {}'.format(metric_type))
+        time.sleep(1)
+        retries_left -= 1
+
+    if not retries_left:
+        raise MetricError('Waiting for metric {} failed. Project resource {}.'.format(metric_type, project_resource))
+    else:
+        return True
+
+
+def create_timeseries_data(metric_type, job_id, project_id, region, environment_name, start, end):
+    return {
+        "metric": {
+            "type": metric_type,
+            "labels": {
+                "job_id": job_id
+            }
+        },
+        "resource": {
+            "type": 'cloud_composer_environment',
+            "labels": {
+                'project_id': project_id,
+                'location': region,
+                'environment_name': environment_name
+            }
+        },
+        "points": [
+            {
+                "interval": {
+                    "startTime": start,
+                    "endTime": end
+                },
+                "value": {
+                    "int64Value": 1
+                }
+            }
+        ]
+    }
+
+
+def increment_counter(client, monitoring_config, metric_type, job_id):
+    start = get_now_rfc3339()
+    end = get_now_rfc3339()
+    timeseries_data = create_timeseries_data(
+        metric_type,
+        job_id,
+        monitoring_config.project_id,
+        monitoring_config.region,
+        monitoring_config.environment_name,
+        start,
+        end)
+    api_create_timeseries(client, monitoring_config, timeseries_data)
+
+
+def increment_job_failure_count(monitoring_config, job_id):
+    try:
+        client = api_client()
+        if not metric_exists(client, monitoring_config.project_resource, BIGGERQUERY_JOB_FAILURE_METRIC_TYPE):
+            api_create_metric(client, monitoring_config.project_resource, BIGGERQUERY_JOB_FAILURE_METRIC)
+            wait_for_metric(client, monitoring_config.project_resource, BIGGERQUERY_JOB_FAILURE_METRIC_TYPE)
+        increment_counter(client, monitoring_config, BIGGERQUERY_JOB_FAILURE_METRIC_TYPE, job_id)
+    except Exception as e:
+        raise MetricError('Cannot increment job failure count: ' + str(e))
```

Comment:
Do you have here full stack trace or only small portion of valuable message ?

Reply:
good point

Reply:
done